pkg/eventpersisterops/persister/evaluation_event.go

```diff
@@ -63,20 +63,30 @@ func NewEvalUserCountUpdater(
 func (u *evalEvtUpdater) UpdateUserCounts(ctx context.Context, evt environmentEventMap) map[string]bool {
 	fails := map[string]bool{}
 	for environmentNamespace, events := range evt {
+		listAutoOpsRules, err := u.listAutoOpsRules(ctx, environmentNamespace)
+		if err != nil {
+			u.logger.Error("failed to list auto ops rules",
+				zap.Error(err),
+				zap.String("environmentNamespace", environmentNamespace),
+			)
+			handledCounter.WithLabelValues(codeFailedToListAutoOpsRules).Inc()
+			// Make sure to retry all the events in the next pulling
+			for id := range events {
+				fails[id] = true
+			}
+			return fails
```

Comment:
If the `encironmentEventMap` contains the two environments' events and some error occurs when executing `listAutoOpsRules` on the processing of the first environment, I assume this `fails` doesn't contain the eventIDs of the second environment. Is it OK?

Reply:
Thanks for pointing that out.
I have fixed the condition.